pkg/generator/caches.go

```diff
@@ -336,3 +367,12 @@ func newExternalEnvoyListenerWithOneCert(ctx context.Context, manager *httpconnm
 
 	return envoy.NewHTTPSListener(config.HTTPSPortExternal, []*v3.FilterChain{filterChain}, enableProxyProtocol)
 }
+
+func newInternalEnvoyListenerWithOneCert(ctx context.Context, manager *httpconnmanagerv3.HttpConnectionManager, kubeClient kubeclient.Interface, enableProxyProtocol bool, certSecretName string) (*v3.Listener, error) {
```

Comment:
The only diff between internal and external is the namespace  and secretaName pair? 
```
os.Getenv(envCertsSecretNamespace) , os.Getenv(envCertsSecretName)
vs
ystem.Namespace(), certSecretName,
``` 
Should we simplify this?

Reply:
Alternatively, rather than making the cert name configurable, you could hard code it, and use the presence / absence of the cert to activate the encryption on the internal interface.

In my experience, this can work fairly well if the component is installed in its own namespace and isn't expected to share. In that case, you could default the secret name to e.g. `kourier-internal-domain-cert` or the like, and then document that if that cert is created, Kourier will use it.

Reply:
Hard cord name is fine, but I assumed that the future plan as:

step-1. Create the secret(=certs) automatically by cert-manager or the library in `knative.dev/pkg`.
step-2. Knative always use TLS traffic with the auto generated certs.

At step-2, `cluster-cert-secrete` in the ConfigMap starts having the default value like:

```
kind: ConfigMap
  name: config-kourier
    cluster-cert-secret: "kourier-internal-domain-cert"
```

And if users want to use their own certs rather than auto-gen certs, they can set the different value in `cluster-cert-secret`.